lib/elixir/lib/exception.ex

```diff
@@ -511,35 +520,49 @@ defmodule SystemLimitError do
 end
 
 defmodule SyntaxError do
-  defexception [file: nil, line: nil, description: "syntax error"]
+  defexception [file: nil, line: nil, description: "syntax error", message: nil]
```

Comment:
It's feels non-intuitive to have `:description` and `:message` at the same time.
What is the purpose for this?